sdk/identity/identity/src/credentials/workloadIdentityCredential.ts

```diff
@@ -123,4 +180,73 @@ export class WorkloadIdentityCredential implements TokenCredential {
     }
     return this.azureFederatedTokenFileContent;
   }
+
+  private async requestOidcToken(
+    oidcRequestUrl: string,
+    systemAccessToken: string
+  ): Promise<string> {
+    logger.info("Requesting OIDC token from Azure Pipelines...");
+    logger.info(oidcRequestUrl);
+
+    const httpClient = createDefaultHttpClient();
+
+    const request = createPipelineRequest({
+      url: oidcRequestUrl,
+      method: "POST",
+      headers: createHttpHeaders({
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${systemAccessToken}`,
+      }),
+    });
+
+    const response = await httpClient.sendRequest(request);
+    const text = response.bodyAsText;
+    if (!text) {
+      throw new CredentialUnavailableError(
+        `${credentialName}: is unavailable. Received null token from OIDC request. Response status = ${response.status}`
+      );
+    }
+    const result = JSON.parse(text);
+    if (result?.oidcToken) {
+      return result.oidcToken;
+    } else {
+      throw new CredentialUnavailableError(
+        `${credentialName}: is unavailable. oidcToken field not detected in the response. Response = ${JSON.stringify(
+          result
+        )}`
+      );
+    }
+  }
+
+  private ensurePipelinesSystemVars(): void {
+    if (
+      process.env.SYSTEM_TEAMFOUNDATIONCOLLECTIONURI &&
+      process.env.SYSTEM_TEAMPROJECTID &&
+      process.env.SYSTEM_PLANID &&
+      process.env.SYSTEM_JOBID &&
+      process.env.SYSTEM_ACCESSTOKEN
+    ) {
+      return;
+    }
+    const missingEnvVars = [];
+    if (!process.env.SYSTEM_TEAMFOUNDATIONCOLLECTIONURI)
+      missingEnvVars.push("SYSTEM_TEAMFOUNDATIONCOLLECTIONURI");
+    if (!process.env.SYSTEM_TEAMFOUNDATIONCOLLECTIONURI)
+      missingEnvVars.push("SYSTEM_TEAMFOUNDATIONCOLLECTIONURI");
+    if (!process.env.SYSTEM_TEAMPROJECTID) missingEnvVars.push("SYSTEM_TEAMPROJECTID");
+    if (!process.env.SYSTEM_PLANID) missingEnvVars.push("SYSTEM_PLANID");
+    if (!process.env.SYSTEM_JOBID) missingEnvVars.push("SYSTEM_JOBID");
+    if (!process.env.SYSTEM_ACCESSTOKEN) missingEnvVars.push("SYSTEM_ACCESSTOKEN");
+    if (missingEnvVars.length > 0)
+      throw new CredentialUnavailableError(
+        `${credentialName}: is unavailable. Missing system variable(s) - ${missingEnvVars.join(
```

Comment:
This isn't a great error message - yes. the variables are missing, but we have a good idea as to why they might be, and what the user should do instead. For the first set, you can generate something like
`The environment variables SYSTEM_JOBID and SYSTEM_PLANID aren't defined - please ensure that you're running this task in an Azure Pipeline.`

We can do more for undefined/non-null check for SYSTEM_ACCESSTOKEN with a message like `Please ensure that the system access token is available in the SYSTEM_ACCESSTOKEN value; this is often most easily achieved by adding a block to the end of your pipeline yaml for the task with:
```yml
env:
- SYSTEM_ACCESSTOKEN: $(System.AccessToken)
```